source/server/http/admin.cc

```diff
@@ -668,7 +714,7 @@ void AdminFilter::onComplete() {
 
   // Under no circumstance should browsers sniff content-type.
   header_map->addReference(headers.XContentTypeOptions, headers.XContentTypeOptionValues.Nosniff);
-  callbacks_->encodeHeaders(std::move(header_map), response.length() == 0);
+  callbacks_->encodeHeaders(std::move(header_map), end_stream && response.length() == 0);
 
   if (response.length() > 0) {
     callbacks_->encodeData(response, true);
```

Comment:
If we want `end_stream` to be general purpose as proposed in the comment above, can we change this call to: 
```
callbacks_->encodeData(response, end_stream);
```
Essentially this would mean that if hystrix or some other callback wanted to send data *and* not end the stream, this would be possible.